libraries/AP_Airspeed/AP_Airspeed_RSC.cpp

```diff
@@ -0,0 +1,420 @@
+// total no of coefficients in each type
+#define RSC_COEFF_T_COL_NO                     4
+// this can be calculated by using the LSB address of the 0th coefficient
+// and the MSB of the 3rd coefficient
+#define RSC_COEFF_ADDRESS_SPACE_SIZE           16
+
+#define TEMPERATURE_RATIO                      0.03125f
+
+#define RSC_DRDY_PIN 18
+
+extern const AP_HAL::HAL &hal;
+
+AP_Airspeed_RSC::AP_Airspeed_RSC(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+bool AP_Airspeed_RSC::init()
+{
+    // get eeprom device
+    _eeprom_dev = hal.spi->get_device("rsc_eeprom");
+    if (!_eeprom_dev) {
+        return false;
+    }
+
+    // get adc device
+    _adc_dev = hal.spi->get_device("rsc_adc");
+    if (!_adc_dev) {
+        return false;
+    }
+
+    // setup data ready pin
+    _drdy_pin = hal.gpio->channel(RSC_DRDY_PIN);
+    _drdy_pin->mode(HAL_GPIO_INPUT);
+
+    // take semaphore
+    if (!_adc_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        return false;
+    }
+
+    // get eeprom contents
+    _get_eeprom_contents();
+
+    // check airspeed type
+    if (strcmp(_pressure_unit, "inH2O") == 0) {
+        _unit = INH2O;
+    }
+
+    // use initial value to setup adc
+    _set_adc_config();
+
+    // set data rate to 175hz
+    _set_data_rate(N_DR_175_SPS);
+
+    // convert temperature first
+    _adc_convert(TEMPERATURE);
+
+    // release semaphore
+    _adc_dev->get_semaphore()->give();
+
+    // timer run at 90hz
+    _adc_dev->register_periodic_callback(1000000UL/90U,
+                                     FUNCTOR_BIND_MEMBER(&AP_Airspeed_RSC::_timer, void));
+    return true;
+}
+
+bool AP_Airspeed_RSC::_data_ready()
+{
+    if (_drdy_pin) {
+        return _drdy_pin->read() != 1;
+    }
+
+    return false;
+}
+
+void AP_Airspeed_RSC::_timer()
+{
+    uint8_t buf[3];
+
+    if (!_data_ready()) {
+        return;
+    }
+
+    // read adc value
+    _adc_read(buf);
+
+    WITH_SEMAPHORE(sem);
+
+    if (_type == TEMPERATURE) {
+        // tempreature is 14 bits in 2s complement representation
+        // each byte is shifted to its position in a 16-bit unsigned integer and from 8 more bits to be left-aligned in a 16-bit integer
+        _t_raw = ((uint16_t)buf[0] << 8) | (uint16_t)buf[1];
+        // right-shift signed integer back to get measurement value
+        _t_raw >>= 2;
+        // apply scaler to get temperature
+        _temperature = _t_raw * TEMPERATURE_RATIO;
+    } else {
+        // pressure is 24 bits in 2s complement representation
+        // each byte is shifted to its position in a 32-bit unsigned integer and from 8 more bits to be left-aligned in a 32-bit integer
+        _p_raw = ((uint32_t)buf[0] << 24) | ((uint32_t)buf[1] << 16) | ((uint32_t)buf[2] << 8);
+        // right-shift signed integer back to get measurement value
+        _p_raw >>= 8;
+        // apply scaler to get pressure
+        if (_unit == INH2O) {
```

Comment:
are other unit options missing here?

Reply:
yes, different unit, ex: Pa, kPa, Mpa, mBar, Bar, inH2O, psi.
![image](https://user-images.githubusercontent.com/16251761/104681521-7d2a4880-572d-11eb-88cd-45fdfa0c0343.png)

If it is other types of sensors, only a few modifications are required, and the driver can work well.